lib/index.ts

```diff
@@ -16,6 +16,19 @@ const cancelableClient: CancelableAxiosInstance = Object.assign(client, {
 	isCancel: Axios.isCancel,
 })
 
+// Add a 401 response interceptor
+cancelableClient.interceptors.response.use((response: AxiosResponse): AxiosResponse     => {
+	if (response.status === 401) {
+		location.reload()
```

Comment:
idea: emit a cancellable event so that apps get a chance to store stuff into local storage (ex: unsent talk messages) before proceeding with the reload

if the event handler cancels the event, the reload is also cancelled

Reply:
that would still mean we force reload any app that doesn't cancel this possibly emitted event

the app is in full control of the http communication. it can catch the 401 and handle it appropriately. the indirection of an event shouldn't be necessary, right?

Reply:
the goal here was to avoid having all apps having to catch 401 in all their possible code paths

if that's too much maybe making it optional and on by default ?

Reply:
Opt-in version submit as https://github.com/nextcloud/nextcloud-axios/pull/537.